Mathlib/Tactic/CategoryTheory/Monoidal.lean

```diff
@@ -566,55 +559,46 @@ structure Result where
 partial def evalComp : NormalExpr → NormalExpr → MonoidalM Result
   | .nil α, .cons β η ηs => do
     let η' := .cons (α.comp β) η ηs
-    try return ⟨η', ← mkAppM ``evalComp_nil_cons #[← α.e, ← β.e, ← η.e, ← ηs.e]⟩
-    catch _ => return ⟨η', mkConst ``True⟩
+    return ⟨η', ← mkAppM ``evalComp_nil_cons #[← α.e, ← β.e, ← η.e, ← ηs.e]⟩
```

Comment:
I don't yet understand what the implications of these changes are and why they are necessary. Could you quickly explain please? Do you know why there used to be a `catch` which isn't needed anymore?

Reply:
Previously I was using `ExprPresenter`, but then the normalization program had trouble generating proof terms for normalization. One reason was that the `MonoidalCategory C` instance in the assumptions was not available. In my understanding, tasks like generating proofs are beyond the scope of `ExprPresenter`.

In the current implementation of `StringDiagram.rpc` function, it explicitly deals with local instances using the `Meta.withLCtx` function.

Reply:
> One reason was that the `MonoidalCategory C` instance in the assumptions was not available. In my understanding, tasks like generating proofs are beyond the scope of `ExprPresenter`.

Ah, no, that was an oversight in ProofWidgets. I pushed a fix for it now, and confirmed that your code also works with `@[expr_presenter]` using updated ProofWidgets.

The real difference between `ExprPresenter`s and the panel widget approach is that the latter is more flexible but needs more boilerplate. With an `ExprPresenter`, your function can only process a single expression when computing the display, as opposed to integrating data from multiple local hypotheses (for instance multiple assumed equalities) in a single display. For this price, you do not have to add extra handling for multiple goals or for visualizing hypotheses.

Since visualizing a single expression is what you are doing now, maybe the limitation is okay? Do you anticipate a need for the flexibility of panel widgets later?

More generally, the `ExprPresenter` API doesn't get used much and I'm not sure if it's even needed. It would be nice to see a real use so that it can be either improved or retired (if we decide it's not useful).

Reply:
@Vtec234  Thanks for explanation! It's helpful. Also, thanks for the update.

I'm not sure if I need such flexibility in the string diagram at the moment. 

My impression of the `ExprPresenter` is that it is nice because it is enough intuitive to use for me.